app/scripts/migrations/106.test.ts

```diff
@@ -0,0 +1,251 @@
+          },
+        },
+      },
+      NameController: {
+        names: {
+          [ACCOUNT_LABEL_NAME_TYPE]: {
+            [ADDRESS_2.toLowerCase()]: {
+              [ACCOUNT_LABEL_CHAIN_ID]: {
+                name: NAME_2,
+                sourceId: 'ens',
+                proposedNames: {},
+              },
+            },
+          },
+        },
+      },
+    };
+
+    const transformedState = await migrate({
+      meta: { version: oldVersion },
+      data: oldState,
+    });
+
+    expect(transformedState.data).toEqual({
+      ...oldState,
+      NameController: {
+        names: {
+          [ACCOUNT_LABEL_NAME_TYPE]: {
+            [ADDRESS_1.toLowerCase()]: {
+              [ACCOUNT_LABEL_CHAIN_ID]: {
+                name: NAME_1,
+                sourceId: null,
+                proposedNames: {},
+              },
+            },
+            [ADDRESS_2.toLowerCase()]: {
+              [ACCOUNT_LABEL_CHAIN_ID]: {
+                name: NAME_2,
+                sourceId: 'ens',
+                proposedNames: {},
+              },
+            },
+          },
+        },
+      },
+    });
+  });
+
+  it('ignores account id entry if existing petname', async () => {
+    const oldState = {
+      OtherController: {},
+      PreferencesController: {
+        identities: {
+          [ADDRESS_1]: {
+            name: NAME_1,
+            address: ADDRESS_1,
+          },
+        },
+      },
+      NameController: {
+        names: {
+          [ACCOUNT_LABEL_NAME_TYPE]: {
+            [ADDRESS_1.toLowerCase()]: {
+              [ACCOUNT_LABEL_CHAIN_ID]: {
+                name: NAME_2,
+                sourceId: 'ens',
+                proposedNames: {},
+              },
+            },
+          },
+        },
+      },
+    };
+
+    const transformedState = await migrate({
+      meta: { version: oldVersion },
+      data: oldState,
+    });
+
+    expect(transformedState.data).toEqual({
+      ...oldState,
+      NameController: {
+        names: {
+          [ACCOUNT_LABEL_NAME_TYPE]: {
+            [ADDRESS_1.toLowerCase()]: {
+              [ACCOUNT_LABEL_CHAIN_ID]: {
+                name: NAME_2,
+                sourceId: 'ens',
+                proposedNames: {},
+              },
+            },
+          },
+        },
+      },
+    });
+  });
+
+  it('ignores account id entry if no name or address', async () => {
+    const oldState = {
+      OtherController: {},
+      PreferencesController: {
+        identities: {
+          [ADDRESS_1]: {
+            name: NAME_1,
+            address: '',
+          },
+          [ADDRESS_2]: {
+            name: '',
+            address: ADDRESS_2,
+          },
+        },
+      },
+    };
+
+    const transformedState = await migrate({
+      meta: { version: oldVersion },
+      data: oldState,
+    });
+
+    expect(transformedState.data).toEqual({
+      ...oldState,
+      NameController: {
+        names: {
+          [ACCOUNT_LABEL_NAME_TYPE]: {},
```

Comment:
Minor, but for the sake of stability and the minimal changes possible, should we not create this `NameController` entry in the state if there have been no entries created?